scripts/reindex.py

```diff
@@ -148,6 +154,16 @@ def main(argv: List[str]):
             parser.error('Cannot specify sources when performing a local reindex')
             assert False
 
+    if args.deindex:
+        require(not any((args.index, args.delete, args.create)),
+                '--deindex is incompatible with --index, --create, and --delete.')
+        require('*' not in source_globs, '--deindex is incompatible with source `*`. '
+                                         'Use --delete instead.')
```

Comment:
Are we 100% sure there are no circumstances in which we would want to remove all data from the indices without destroying them? We have the option to `--create` the indices without populating them, so de-populating them without `--delete`ing seems like it would make sense for symmetry.

Reply:
@hannes-ucsc didn't seem to think so. The use-case this script option is designed for is deleting a single source (in order to replace it with a different one usually). The issue is that * is the default for sources. We wanted to prevent someone using the `--deindex` option from accidentally deleting everything. This `require()` ensures that a `--source` is specified.

If for some reason we wanted empty indexes, the fastest/most efficient way to do this would be to delete and then create anew. Using the `--deindex` option is much slower since it performs a search over all of the documents.

Reply:
Makes sense, I appreciate the explanation